src/math/exp_fcn.c

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+
+/* Include Files */
+#include <sof/math/exp_fcn.h>
+#include <sof/math/numbers.h>
+#include <sof/common.h>
+#include <rtos/bit.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#define BIT_MASK_Q62P2 0x4000000000000000
+#define CONVERG_ERROR 28823037607936LL	// error smaller than 1e-4,1/2 ^ -44.7122876200884
+#define BIT_MASK_LOW_Q27P5 0x8000000
+#define QUOTIENT_SCALE BIT(30)
+#define TERMS_Q23P9 8388608
+#define LSHIFT_BITS 8192
+
+/* inv multiplication lookup table */
+/* LUT = ceil(1/factorial(b_n) * 2 ^ 63) */
+const int64_t iv_t[19] = {
```

Comment:
`iv_t` is easier to grep than `iv` so this is progress but it's still a very short and super cryptic name for a global constant.